### PR TITLE
Add documentation to class and methods

### LINKED ISSUE #28: add documentation to class and methods

### PR DESCRIPTION
**Pull request by pcolaianni**
closes #28

### CODE REVIEW

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 1]
```diff
@@ -52,12 +52,20 @@
 namespace OpenMS
 {
   /**
-  @brief TODO BRIEF EXPLANATION HERE
-
-  TODO MULTILINE
-  EXPLANATION
-  HERE
-
+    @brief This class implements algorithms that, given an experiment and a target
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: > This class filters, annotates, picks, and scores spectra (e.g., taken from a DDA experiment) based on a target list.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 2]
```diff
@@ -110,64 +118,129 @@ namespace OpenMS
...
 
+    /**
+      Picks a spectrum's peaks and saves them in picked_spectrum.
+      The spectrum is first smoothed with a Gaussian filter (default) or using the
+      Savitzky-Golay method.
+      For these filters it is possible to set parameters like gauss_width_,
+      sgolay_frame_length_ and sgolay_polynomial_order_ through their setters.
+      The peak picking is executed with PeakPickerHiRes. It's possible to set the
+      signal_to_noise_ parameter.
+      @param[in] spectrum The input spectrum
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: I would remove the [in] and [out].  I am not sure doxygen is able to interpret that correctly.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 3]
```diff
@@ -110,64 +118,129 @@ namespace OpenMS
 
     void getDefaultParameters(Param& params);
 
+    /**
+      Picks a spectrum's peaks and saves them in picked_spectrum.
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: You will need a @brief to start off the description similar to the class doc strings.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h (hunk 2 above)
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: There should also be a line break between the @brief, longer description, and parameters.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 4]
```diff
@@ -110,64 +118,129 @@ namespace OpenMS
...
+    */
     void annotateSpectra(
       const std::vector<MSSpectrum>& spectra,
       const TargetedExperiment& targeted_exp,
       std::vector<MSSpectrum>& annotated_spectra,
       FeatureMap& features
     );
 
+    /**
+      Assigns a score to the spectra given in input and saves them in scored_spectra.
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: > ...given an input...

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 5]
```diff
@@ -110,64 +118,129 @@ namespace OpenMS
...
+    */
     void extractSpectra(
       const PeakMap& experiment,
       const TargetedExperiment& targeted_exp,
       std::vector<MSSpectrum>& extracted_spectra,
       FeatureMap& extracted_features
     );
 
+    /**
+      Given a list of spectra, it selects one spectrum for each possible annotation (transition name).
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: > The method selects the highest scoring spectrum for each possible annotation (i.e., transition name)

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 6]
```diff
@@ -110,64 +118,129 @@ namespace OpenMS
...
+      The choice is based on the spectrum's score.
+      Also the related features get filtered.
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: 196 and 197 can be removed

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 7]
```diff
@@ -110,64 +118,129 @@ namespace OpenMS
...
       const std::vector<MSSpectrum>& scored_spectra,
       std::vector<MSSpectrum>& selected_spectra,
       const FeatureMap& features,
       FeatureMap& selected_features
     );
 
+    /**
+      Given a list of spectra, it selects one spectrum for each possible annotation (transition name).
+      The choice is based on the spectrum's score.
+      @param[in] scored_spectra Input annotated and scored spectra
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: Same as above.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 8]
```diff
@@ -110,64 +118,129 @@ namespace OpenMS
...
-    UInt sgolay_polynomial_order_;
-    double gauss_width_;
-    bool use_gauss_;
-    double signal_to_noise_;
-
-    // score
-    double tic_weight_;
-    double fwhm_weight_;
-    double snr_weight_;
+    double mz_tolerance_; /**< Precursor MZ tolerance used during the annotation phase */
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h:239]: Nice explanations.

File: src/openms/source/ANALYSIS/OPENSWATH/SpectrumExtractor.cpp [hunk 9]
```diff
@@ -377,19 +377,19 @@ namespace OpenMS
...
-      sne.init(annotated[i].begin(), annotated[i].end());
+      sne.init(annotated_spectra[i].begin(), annotated_spectra[i].end());
       double avgSNR = 0.0;
-      for (it=annotated[i].begin(); it!=annotated[i].end(); ++it)
+      for (it=annotated_spectra[i].begin(); it!=annotated_spectra[i].end(); ++it)
       {
         avgSNR += sne.getSignalToNoise(it);
       }
-      avgSNR /= annotated[i].size();
+      avgSNR /= annotated_spectra[i].size();
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/OPENSWATH/SpectrumExtractor.cpp:386]: Good idea with the name change.

**Review by dmccloskey — COMMENTED** [review]
I made a few minor suggestions for the documentation.  It is clear to me what the class/methods/members do so hopefully it will be clear to the guys as OpenMS

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h (hunk 2 above)
Comment (pcolaianni) [reply src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: The behaviour is documented here: http://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdparam
Checking the source code, nobody is using it. Should I remove it?

**Comment by pcolaianni** [comment]
@dmccloskey thanks for the review. Once you decide on my comment about [in] and [out], I'll also apply that and then merge.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h (hunk 2 above)
Comment (dmccloskey) [reply src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: I did not realize that.  Go ahead and leave it in then.  We can see what they say on the PR.

**Pull request merged**